fixedwidth/src/main/java/org/apache/metamodel/fixedwidth/FixedWidthLineParser.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.fixedwidth;
+
+import java.io.IOException;
+import java.text.CharacterIterator;
+import java.text.StringCharacterIterator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FixedWidthLineParser {
+ 
+    private final int _fixedValueWidth;
+    private final int[] _valueWidths;
+    private final boolean _failOnInconsistentLineWidth;
+    private final int _expectedLineLength;
+    private final boolean _constantWidth;
```

Comment:
The 5 above fields are all represented with one `FixedWidthConfiguration` object. It's not important to me, but I think it could reduce complexity for you to reuse that class instead of calling out these individual fields.